### PR TITLE
update enums

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

[Fixes this](https://github.com/OpsLevel/terraform-provider-opslevel/issues/391)

## Changelog

Update enums

- [X] List your changes here
- [X] Make a `changie` entry

## Tophatting

`task ci` works

### CODE REVIEW

File: enum.go [hunk 1]
```diff
@@ -270,6 +284,7 @@ const (
 	CheckTypeGitBranchProtection CheckType = "git_branch_protection" // Verifies that all the repositories on the service have branch protection enabled.
 	CheckTypeServiceDependency   CheckType = "service_dependency"    // Verifies that the service has either a dependent or dependency.
 	CheckTypeRepoGrep            CheckType = "repo_grep"             // Run a comprehensive search across the service's repository using advanced search parameters.
+	CheckTypePackageVersion      CheckType = "package_version"       // Verifies certain aspects of a service using or not using software packages.
```
Comment (rocktavious) [inline enum.go:287]: can you leave this one off for now - we don't support this check yet

File: enum.go [hunk 2]
```diff
@@ -868,16 +972,20 @@ var AllToolCategory = []string{
 type UserRole string
 
 const (
-	UserRoleUser      UserRole = "user"       // A regular user on the account.
-	UserRoleAdmin     UserRole = "admin"      // An administrator on the account.
-	UserRoleBasicUser UserRole = "basic_user" // A basic user on the account with limited access.
+	UserRoleUser           UserRole = "user"            // A regular user on the account.
+	UserRoleAdmin          UserRole = "admin"           // An administrator on the account.
+	UserRoleBasicUser      UserRole = "basic_user"      // A basic user on the account with limited access.
```
Comment (rocktavious) [inline enum.go:977]: `basic_user` is no more - why is it still here?

File: enum.go [hunk 3]
```diff
@@ -462,6 +478,94 @@ var AllHasDocumentationTypeEnum = []string{
 	string(HasDocumentationTypeEnumAPI),
 }
 
+// PackageConstraintEnum represents possible values of a package version check constraint.
+type PackageConstraintEnum string
```
Comment (rocktavious) [inline enum.go:482]: this one should be left out till we do the package check work in next sprint

File: enum.go (hunk 2 above)
Comment (davidbloss) [reply enum.go:977]: Good question, I'll remove it

**Review by taimoorgit — APPROVED** [review]
Should also add a `Removed` entry for `UserRoleBasicUser`/`basic_user`.

**Pull request merged**